File: java_refactoring/src/main/java/org/example/solution/TaxInfo.java
package org.example.solution;

public class TaxInfo {
    private final double salesTax;
    private final double propertyTax;
    private final double incomeTax;
    private final double totalTax;

    public TaxInfo(double totalValue) {
        this.salesTax = totalValue * .1 ;
        this.propertyTax = totalValue * .2;
        this.incomeTax = totalValue * .3;
        this.totalTax = totalValue * .6;
    }

    public double getTotalTax() {
        return totalTax;
    }
    public double getPropertyTax() {
        return propertyTax;
    }

    @Override
    public String toString() {
        return "sales: " + salesTax + ":"
                + "property: " + propertyTax + ":"
                + "income: " + incomeTax + ":"
                + "total: " + totalTax;
    }


}

File: typescript_refactoring/example/solution/TaxInfo.ts
export interface TaxInfo {
    salesTax: number;
    propertyTax: number;
    incomeTax: number;
    totalTax: number;
}

export function calculateTaxes(totalValue: number): TaxInfo {
    return {
        salesTax: totalValue * 0.1,
        propertyTax: totalValue * 0.2,
        incomeTax: totalValue * 0.3,
        totalTax: totalValue * 0.6,
    };
}
